src/applications/edu-benefits/components/AutosuggestField.jsx

```diff
@@ -0,0 +1,296 @@
+
+      // If this is an non-object field then the label will
+      // be included by ReviewFieldTemplate
+      if (schema.type !== 'object') {
+        return readOnlyData;
+      }
+
+      return (
+        <div className="review-row">
+          <dt>{this.props.uiSchema['ui:title']}</dt>
+          <dd>{readOnlyData}</dd>
+        </div>
+      );
+    }
+
+    return (
+      <div>
+        <Downshift
+          onChange={this.handleChange}
+          onInputValueChange={this.handleInputValueChange}
+          inputValue={this.state.input}
+          selectedItem={this.state.input}
+          onOuterClick={this.handleBlur}
+          itemToString={item => {
+            if (typeof item === 'string') {
+              return item;
+            }
+
+            return item.label;
+          }}
+          render={({
+            clearSelection,
+            getInputProps,
+            getItemProps,
+            isOpen,
+            highlightedIndex
+          }) => (
+            <div className="edu-complaint-input-controls">
+              <input {...getInputProps()}/>
+              {isOpen && (loading || items) ? (
+                <div className="ds-u-border--1 ds-u-padding--1 ds-c-autocomplete__list">
+                  {listLabel && (
+                    <h5
+                      className="ds-u-margin--0 ds-u-padding--1"
+                      id={this.labelId}>
+                      {listLabel}
+                    </h5>
+                  )}
+
+                  <ul
+                    aria-labelledby={listLabel ? this.labelId : null}
+                    className="ds-c-list--bare"
+                    id={this.listboxId}
+                    role="listbox">
+                    {!loading && this.state.suggestions
+                      .map(({ label: itemLabel, value: itemValue }, index) => (
+                        <li
+                          aria-selected={highlightedIndex === index}
+                          className={
+                            highlightedIndex === index
+                              ? 'ds-c-autocomplete__list-item ds-c-autocomplete__list-item--active'
+                              : 'ds-c-autocomplete__list-item'
+                          }
+                          key={index}
+                          role="option"
+                          {...getItemProps({
+                            item: {
+                              label: itemLabel,
+                              value: itemValue
+                            }
+                          })}>
+                          {itemLabel}
+                        </li>
+                      ))}
+                  </ul>
+                </div>
+              ) : null}
+
+              <button
+                aria-label={'clear all fields on this page'}
+                className="ds-u-float--right ds-u-padding-right--0"
+                onClick={e => this.handleClearSelection(e, clearSelection)}>
```

Comment:
One thing I noticed in all three places we're using Downshift is that we don't make full use of the library. Fex Downshift provides a `clearSelection()` callback that clears the input and sets the expected focus.